controllers/heat_controller.go

```diff
@@ -373,6 +374,42 @@ func (r *HeatReconciler) reconcileNormal(ctx context.Context, instance *heatv1be
 	//
 	// normal reconcile tasks
 	//
+	keystoneAPI, err := keystonev1.GetKeystoneAPI(ctx, helper, instance.Namespace, map[string]string{})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// get the password of the service user from the secret
+	password, _, err := oko_secret.GetDataFromSecret(
+		ctx,
+		helper,
+		instance.Spec.Secret,
+		10,
+		instance.Spec.PasswordSelectors.Service)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	//
+	// get admin authentication OpenStack
+	//
+	os, _, err := keystonev1.GetAdminServiceClient(
+		ctx,
+		helper,
+		keystoneAPI,
+	)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	_, err = os.CreateUser(
```

Comment:
we also have to create heat_stack domain and assign admin role between heat_stack and this heat_stack_domaiN_admin user

Reply:
I think I'll need to add a new Domain package to lib-common to do this. Rather than manually using doing it here. I'll work on the lib-common patch for it.

Reply:
For reference:
https://github.com/openstack-k8s-operators/lib-common/pull/224

Reply:
where do we plan to delete this user? I think for now its ok. do do it like this. I was planning to split the keystoneservice CRD as this right now creates the user and the service. we probably want to have a crd for user and service so that we use the same interface. maybe we then should have there also a keystonedomain crd? And like we do with the service and endpoint the user/domain then gets cleaned up. Does that make sense?